sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -106,6 +112,34 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invokedMethod = c.SemanticModel.GetSymbolInfo(c.Node).Symbol as IMethodSymbol;
+                    if (invokedMethod == null)
+                    {
+                        return;
+                    }
+
+                    if (invokedMethod.IsInType(KnownType.System_Uri))
+                    {
+                        return;
+                    }
+
+                    var stringUrlParams = GetStringUrlParamIndexes(invokedMethod);
+                    if (!stringUrlParams.Any())
```

Comment:
Performance difference between a single call vs property access would be marginal, so we should prefer readability.
"stringUrlParams.Count > 0" is more readable to me as well (as there is no negation). Will update.